main.tf

```diff
@@ -85,14 +107,10 @@ resource "google_service_account_iam_binding" "bastion_sa_user" {
 }
 
 resource "google_project_iam_member" "bastion_sa_bindings" {
-  for_each = toset(compact(concat(
-    var.service_account_roles,
-    var.service_account_roles_supplemental,
-    ["projects/${var.project}/roles/${google_project_iam_custom_role.compute_os_login_viewer.role_id}"]
-  )))
+  count = local.bastion_roles_count
```

Comment:
Instead of requiring a switch to using `count` and custom logic, what if we simply separated the custom role assignment into a separate (single) resource? Then we could leave this resource alone and simplify the logic.

Reply:
I'll give it a shot.  I didn't deconstruct things to determine if it is the custom role at the end of the list which is causing the problem, or if it is the other 2 lists.  I'll play around with your suggestion tomorrow and see if that fixes it.  I'll push an update or post a comment here that it didn't work.

Reply:
Yep.  That cleaned it up.

Reply:
Great! It's much simpler now, thanks for looking into it.